webboot/webboot.go

```diff
@@ -23,9 +40,88 @@ var (
 	ipv6    = flag.Bool("ipv6", true, "use IPV6")
 )
 
+func wgetArgs(argURL string) (read io.ReadCloser, outFile string, werr error) {
+	//checks if argURL is a key in osSystems map
+	if _, ok := osSystems[argURL]; !ok {
+		//if argURL is not the map, use it as a URL link for wget
+		read, werr = wget(argURL)
+	} else {
+		// if argURL is in osSystems map, use its download link for wget
+		read, werr = wget(osSystems[argURL].DownloadLink)
+		outFile = argURL
+	}
+	return read, outFile, werr
+}
+
+func wget(arg string) (io.ReadCloser, error) {
+	resp, err := http.Get(arg)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+func writer(read io.ReadCloser, fileName string) (string, error) {
```

Comment:
// func writer copies from an io.ReadCloser to a named file

Reply:
I provided a brief comment explaining what each function does besides the basic wget function since wget is self explanatory.

Reply:
Yes, I agree how a comment would be nice to explain what the writer function does. I provided a brief comment explaining its implementations.